ivy/functional/backends/torch/general.py

```diff
@@ -41,9 +41,13 @@ def to_numpy(x: torch.Tensor, copy: bool = True) -> np.ndarray:
         else:
             return x
     elif torch.is_tensor(x):
-        if x.dtype is torch.bfloat16:
-            x = x.to(torch.float16)
         if copy:
+            if x.dtype is torch.bfloat16:
+                if ivy.default_float_dtype(as_native=True) is torch.bfloat16:
```

Comment:
you should store the return of `ivy.default_float_dtype(as_native=True)` above this line, otherwise we can end up calling this same line twice.